### PR TITLE
Add design doc for optimized platform view layers

### PR DESCRIPTION
**Pull request by knopp**
Add link to design doc.

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -328,6 +328,7 @@
       { "source": "/go/nullable-cupertinothemedata-brightness", "destination": "https://docs.google.com/document/d/1qivq0CdkWGst5LU5iTLFUe_LTfLY84679-NxWiDgJXg/edit", "type": 301 },
       { "source": "/go/ondirtycallbacks", "destination": "https://docs.google.com/document/d/1Vk_QWC92fFGxx2oIrIjkCL0ZZFHxmrprlLedyQsnkus/edit?usp=sharing", "type": 301 },
       { "source": "/go/opengl-on-ios", "destination":"https://docs.google.com/document/d/1kvrb6HeTRN4noAKO82o6x-Ii61PW-BQ-8s9YNtLgosM", "type": 301 },
+      { "source": "/go/optimized-platform-view-layers", "destination": "https://docs.google.com/document/d/1YHwVz7-F03psEzHxByGka_lIFaDV5K90BJMbNxQeK4k/edit#", "type": 301 },
```
Comment (loic-sharma) [inline firebase.json:331]: Should this doc be public?

Comment (knopp) [reply firebase.json]: My bad, the URL was incomplete. Can you retry?

Comment (sfshaza2) [reply firebase.json]: I tried opening the [url](https://docs.google.com/document/d/1YHwVz7-F03psEzHxByGka_lIFaDV5K90BJMbNxQeK4k/edit) in an incognito window and it doesn't work for me. It requires a login. I can't land until this opens in an incognito window.

Comment (knopp) [reply firebase.json]: Did you try URL from [this commit](https://github.com/flutter/website/pull/7866/commits/10f9857cef7f7e51852ce08cba0b7bb3ba8b93f1)? It works for me in incognto mode. The one before (from previous commit, mentioned here), doesn't.

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**